.github/workflows/main.yml

```diff
@@ -204,7 +203,25 @@ jobs:
         run: |
           dart pub upgrade
           dart test
-
+  test_flutter:
+    # strategy:
+    #   matrix:
+    #     version: ["3.16.0", "3.19.0", "3.21.0"]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          # flutter-version: ${{ matrix.version }}
+          channel: stable
+          cache: true
+          cache-key: "flutter-:os:-:channel:-:version:"
+      - name: Install dependencies in drift/example/app
+        run: |
+          cd examples/app
```

Comment:
Can we still apply the `prepare` action here? That one caches a lot of pub dependencies so it might speed up the CI.

Reply:
This `subosito/flutter-action@v2` caches by itself

Reply:
Take a look here https://github.com/subosito/flutter-action?tab=readme-ov-file#caching

This action installs flutter. We could replace the dart setup with flutter setup, but that would mean we could never run any actions on versions of dart that aren't part of a flutter release.

Reply:
I meant package dependencies from pub.dev which are cached from `~/.pub-cache` across runs - I agree with using dart setup for most of the CI and flutter setup here.

Reply:
I think those are cached too

https://github.com/subosito/flutter-action/blob/44ac965b96f18d999802d4b807e3256d5a3f9fa1/action.yaml#L103-L111

Reply:
Sure - is there an actions run with this I could take a look at to see where it went wrong? I couldn't find one from a quick look, but I can also just try it tomorrow.

Reply:
> Sure - is there an actions run with this I could take a look at to see where it went wrong?  
What's an "actions run"? I'm not understanding 🤷

The only part here that isn't currently working is the cache for pub.
I'm assuming whatever you did for the main dart cache can be copied for here.
The cache for the entire flutter sdk works great.

Reply:
`pub-cache-key` works, but it just seems to sensitive to changes

Tip: Install `act` and `github-act-cache-server`, makes this much easier than committing empty and pushing to github

Reply:
> What's an "actions run"? I'm not understanding 🤷

Sorry, I meant a workflow run of the GitHub action as you've configured it.

>  The cache for the entire flutter sdk works great.

Actually, `flutter-action` also sets up a cache for pub dependencies by itself. So I think we don't need to use our custom cache here, we can leave it as it is.